src/docs/components/properties.md

```diff
@@ -312,9 +312,46 @@ export class MyHttpService {
 
 In TypeScript, `MyHttpService` is both an `Object` and a 'type'. When using user-defined types like `MyHttpService`, the
 type must always be exported using the `export` keyword where it is declared. The reason for this is Stencil needs to
-know what type the prop `myHttpService` is when passing an instance of `MyHttpService` to `TodoList` from a parent
+know what type the prop `httpService` is when passing an instance of `MyHttpService` to `TodoListItem` from a parent
 component.
 
+To set `httpService` in TSX, assign the prop name in the custom element's tag to the desired value like so:
+```tsx
+// TodoList.tsx
+import { Component, h } from '@stencil/core';
+import { MyHttpService } from '../MyHttpService';
+
+@Component({
+   tag: 'todo-list',
+   styleUrl: 'todo-list.css',
+   shadow: true,
+})
+export class ToDoList {
+   private httpService = new MyHttpService();
+
+   render() {
+      return <todo-list-item httpService={this.httpService}></todo-list-item>;
+   }
+}
+```
+Note that the prop name is using `camelCase`, and the value is surrounded by curly braces.
+
+It is not possible to set `Object` props via an HTML attribute like so:
+```html
+<!-- this will not work -->
+<todo-list-item http-service="{ /* implementation omitted */ }"></todo-list-item>
+```
+The reason for this is that Stencil will not attempt to serialize object-like strings written in HTML into a JavaScript object.
+Doing so can be expensive at runtime, and runs the risk of losing references to other nested JavaScript objects.
```

Comment:
This in turn is part of a limitation of HTML attributes right? Might be good just to say something like

```suggestion
This is because HTML attribute values must be strings, and Stencil does not have support
for automatically deserializing JSON (or any other data interchange format) from attributes.
```

is that accurate?

Reply:
It's accurate, although Stencil tends to muddy the waters by serializing numbers represented as strings to `type number` in the JS (and TS) type system, and the same for booleans. 

Instead of getting into that (more than we have in the number/boolean sections), I wanted to call out a problem I've seen in Slack. Specifically, I've seen folks try to do something like:
```html
<to-do-list-item todos="{'thing': 'clean garage', 'isComplete': 'false'}"></to-do-list-item>
```
in HTML. Thoughts?

Reply:
Ok gotcha, so maybe we could leave off any mention of HTML attributes needing to be strings, but call out something along the lines of 'Stencil does not have any support for deserializing objects from JSON' ?

Reply:
👍 Done in 57415a573